src/payload.ts

```diff
@@ -0,0 +1,394 @@
+interface StorageData {
+	apiUrl: string
+}
+
+//Make sure the payload is injected only once per context at most
+// @ts-ignore
+if (window.SECRET_EMOJI_KEY != "set") {
+	// @ts-ignore
```

Comment:
love the typescript conversion. its a shame the inconsistent spacing makes the diff weird